src/main/java/com/emc/pravega/perf/WriterWorker.java

```diff
@@ -55,33 +57,79 @@
 
     @Override
     public Void call() throws InterruptedException, ExecutionException {
-        CompletableFuture retFuture = null;
-        Random rand = new Random();
-
-        for (int i = 0; (i < eventsPerWorker) &&
-                (Duration.between(StartTime, Instant.now()).getSeconds() < secondsToRun); i++) {
-
-            // Construct event payload
-            String val = System.currentTimeMillis() + ", " + workerID + ", " + (int) (Math.random() * 200);
-            String payload = String.format("%-" + messageSize + "s", val);
-            String key;
-            if (isRandomKey) {
-                key = Integer.toString(workerID + rand.nextInt());
-            } else {
-                key = Integer.toString(workerID);
+        perf.benchmark();
+        return null;
+    }
+
+    private class eventsWriter implements performance {
+
+        public void benchmark() throws InterruptedException, ExecutionException {
+            CompletableFuture retFuture = null;
+            Random rand = new Random();
+
+            for (int i = 0; i < events; i++) {
+
+                // Construct event payload
+                String val = System.currentTimeMillis() + ", " + workerID + ", " + (int) (Math.random() * 200);
+                String payload = String.format("%-" + messageSize + "s", val);
+                String key;
+                if (isRandomKey) {
+                    key = Integer.toString(workerID + rand.nextInt());
+                } else {
+                    key = Integer.toString(workerID);
+                }
+
+                final Instant startTime = Instant.now();
+                retFuture = writeData(key, payload);
+                // event ingestion
+                retFuture = stats.recordTime(retFuture, startTime, payload.length());
             }
 
-            final Instant startTime = Instant.now();
-            retFuture = writeData(key, payload);
-            // event ingestion
-            retFuture = stats.recordTime(retFuture, startTime, payload.length());
+            flush();
+
+            //Wait for the last packet to get acked
```

Comment:
A call to flush on an event stream writer, will block until the last packet is acked.